dash/App/Tokens/AddToken/index.js

```diff
@@ -383,25 +399,58 @@ class AddToken extends Component {
       tokenData: {
         name: '',
         symbol: '',
-        decimals: ''
-      }
+        decimals: '',
+        totalSupply: ''
+      },
+      error: null
     }
   }
 
   async updateTokenData(contractAddress, chainId) {
-    const { name, symbol, decimals } = await link.invoke('tray:getTokenDetails', contractAddress, chainId)
-    this.setState({ tokenData: { name, symbol, decimals } })
+    const isValidAddress = isAddress(contractAddress)
```

Comment:
let's separate these out into 2 functions, checking for a valid address and updating the token data is 2 different operations